src/scenes/base.ts

```diff
@@ -12,9 +11,7 @@ export interface SceneOptions<C extends Context> {
   leaveHandlers: ReadonlyArray<MiddlewareFn<C>>
 }
 
-export class BaseScene<
-  C extends SceneContext = SceneContext
-> extends Composer<C> {
+export class BaseScene<C extends Context = Context> extends Composer<C> {
```

Comment:
Shouldn't `C` extend a more specific subtype here?

Reply:
I mean, some of the scenes types (I don't know which) require `ctx.session` to be present, types should express that, no?

Reply:
In theory yes, but this has proven not to be practical. `ctx.session` should contain `_session` which in turn contains a custom type, so we'd immediately need another type parameter here that has to match with `C`. In addition, there is mutual recursion (`ctx` depends on `ctx.scene` depends on `ctx`), hence making extending any of the two extremely challenging.

I can write a more verbose explanation why `Context` is the best constraint here, for now I'll just say that I tried to apply a few different more advanced types. I had to revert all of the changes because it became incomprehensible and messy.